gwpy/types/series.py

```diff
@@ -1024,7 +1024,7 @@ def crop(self, start=None, end=None, copy=False):
                 side="left",
             )
         else:
-            idx1 = floor((xtype(end) - x0) / self.dx.value)
+            idx1 = round((xtype(end) - x0) / self.dx.value)
```

Comment:
See below, but slicing is done up to idx1, not inclusive.

So if the division results in 10.99999999, where it is really 11, it will be rounded down to 10. Since the grid is regular, there should be no points where we have e.g. 10.5. They should be floating representation of ints. See the unit test for an example.